src/main/java/com/google/devtools/build/lib/sandbox/cgroups/Mount.java

```diff
@@ -0,0 +1,53 @@
+package com.google.devtools.build.lib.sandbox.cgroups;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Files;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@AutoValue
+public abstract class Mount {
+    /**
+     * A regexp that matches cgroups entries in {@code /proc/mounts}.
+     *
+     * The format is documented in https://man7.org/linux/man-pages/man5/fstab.5.html
+     */
+    private static final Pattern CGROUPS_MOUNT_PATTERN =
+        Pattern.compile("^[^\\s#]\\S*\\s+(?<file>\\S*)\\s+(?<vfstype>cgroup2?)\\s+(?<mntops>\\S*).*");
+
+    public abstract Path path();
+    public abstract String type();
+    public abstract List<String> opts();
+    public boolean isV2() {
+        return type().equals("cgroup2");
+    }
+
+    static Mount create(Path path, String type, List<String> opts) {
+        return new AutoValue_Mount(path, type, opts);
+    }
+
+    static List<Mount> parse(File procMounts) throws IOException {
```

Comment:
nit: Document what this method does, i.e. reads /proc/mounts and returns `Mount(s)` of cgroup dirs.

Reply:
Done